compiler-plugin-test/src/test/java/io/ballerina/stdlib/persist/compiler/plugin/CompilerPluginTest.java

```diff
@@ -57,244 +57,609 @@ private Package loadPackage(String path) {
+                },
+                new String[]{
+                        DiagnosticsCodes.PERSIST_124.getCode()
+                },
+                1);
     }
 
     @Test
     public void testEntityClosedRecord2() {
-        // todo: test all thrown errors.
-        testDiagnostic("package_31", "the entity 'MedicalNeed2' should be a closed record.",
-                DiagnosticsCodes.PERSIST_124.getCode(), 3, 1);
-    }
-
-    @Test
-    public void testEntityClosedRecord3() {
-        testDiagnostic("package_12", "'reference' field only allows inline initialisation",
-                DiagnosticsCodes.PERSIST_127.getCode(), 8, 1);
-    }
+        List<Diagnostic> errorDiagnosticsList = getDiagnostic("package_31");
+        testDiagnostic(
+                errorDiagnosticsList,
+                new String[]{
+                        "unsupported features: in-line record type is not supported",
+                        "the entity 'MedicalNeed2' should be a closed record",
+                        "unsupported features: in-line record type is not supported"
+                },
+                new String[]{
+                        DiagnosticsCodes.PERSIST_121.getCode(),
+                        DiagnosticsCodes.PERSIST_124.getCode(),
+                        DiagnosticsCodes.PERSIST_121.getCode()
+                },
+                3);
+    }
+
+    @Test
+    public void testFieldInitialisation() {
+        List<Diagnostic> errorDiagnosticsList = getDiagnostic("package_12");
+        testDiagnostic(
+                errorDiagnosticsList,
+                new String[]{
+                        "'keyColumns' field only allows inline initialisation",
+                        "'reference' field only allows inline initialisation",
+                        "'key' field only allows inline initialisation",
+                        "'uniqueConstraints' field only allows inline initialisation",
+                        "'tableName' field only allows inline initialisation",
+                        "invalid initialization: auto increment is only allowed for primary key field",
+                        "'startValue ' field only allows inline initialisation",
+                        "'increment' field only allows inline initialisation"
+                },
+                new String[]{
+                        DiagnosticsCodes.PERSIST_127.getCode(),
+                        DiagnosticsCodes.PERSIST_127.getCode(),
+                        DiagnosticsCodes.PERSIST_127.getCode(),
+                        DiagnosticsCodes.PERSIST_127.getCode(),
+                        DiagnosticsCodes.PERSIST_127.getCode(),
+                        DiagnosticsCodes.PERSIST_108.getCode(),
+                        DiagnosticsCodes.PERSIST_127.getCode(),
+                        DiagnosticsCodes.PERSIST_127.getCode()
+                },
+                8);
+    }
+
+    private List<Diagnostic> getDiagnostic(String packageName) {
```

Comment:
Shall we do an error count assert here? Else testDiagnostic can lead to index out of bounds